File: dotnet/DesignPatterns/CreationalDesignPatterns/FactoryDesignPattern/FactoryDesignPattern.Example1/Concretes/EmailNotificationService.cs
using FactoryDesignPattern.NotificationAPI.Abstracts;
using FactoryDesignPattern.NotificationAPI.Models;

namespace FactoryDesignPattern.NotificationAPI.Concretes;

public class EmailNotificationService : INotificationService
{
	public string Send(string to, string message)
	{
		string result = $"Sending Email to {to} with message: {message}";
		Console.WriteLine(result);

		return result;
	}
}

File: java/DesignPatterns/CreationalDesignPatterns/FactoryDesignPattern/FactoryDesignPattern.Example1/src/main/java/creational/factory/example1/services/concretes/EmailNotificationService.java
package creational.factory.example1.services.concretes;

import creational.factory.example1.services.abstracts.INotificationService;

public class EmailNotificationService implements INotificationService {
    public String send(String to, String message) {
        String result = "Sending Email to " + to + " with message: " + message;
        System.out.println(result);
        return result;
    }
}
